e2e/tests/upgrades/v6/upgrade_test.go

```diff
@@ -0,0 +1,184 @@
+package v6
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/strangelove-ventures/ibctest/v6/chain/cosmos"
+	"github.com/strangelove-ventures/ibctest/v6/ibc"
+	"github.com/strangelove-ventures/ibctest/v6/test"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/e2e/testconfig"
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+)
+
+const (
+	haltHeight         = uint64(100)
+	blocksAfterUpgrade = uint64(10)
+)
+
+func TestUpgradeV6TestSuite(t *testing.T) {
```

Comment:
there are some assumptions being made now that all upgrade tests live under the `TestUpgradeTestSuite` function. See [this workflow](https://github.com/cosmos/ibc-go/blob/5f16e07990f124b728f51c2beb1a8aeb1a7ff1a1/.github/workflows/e2e-upgrade.yaml#L23).

There's nothing stopping us from changing this, just highlighting this as the upgrade tests won't automatically start running this with the current structure.

Reply:
On a related note, currently, we **don't** run upgrade tests on PRs, this is achieved by explicitly excluded them. We do that

[here for forks](https://github.com/cosmos/ibc-go/blob/5f16e07990f124b728f51c2beb1a8aeb1a7ff1a1/.github/workflows/e2e-fork.yml#L27) and [here for regular prs](https://github.com/cosmos/ibc-go/blob/036c09e632ea92b39e25c562037d30bca316a64e/.github/workflows/e2e.yaml#L51)